include/s3select.h

```diff
@@ -651,9 +652,9 @@ struct s3select : public bsc::grammar<s3select>
 
             mulldiv_operand = arithmetic_argument | ('(' >> (arithmetic_expression) >> ')') ; 
 
-            list_of_function_arguments = (arithmetic_expression)[BOOST_BIND_ACTION(push_function_arg)] >> *(',' >> (arithmetic_expression)[BOOST_BIND_ACTION(push_function_arg)]) ;
-
-            function =( (variable >>  '(' )[BOOST_BIND_ACTION(push_function_name)]   >> list_of_function_arguments >> ')' ) [BOOST_BIND_ACTION(push_function_expr)];
+            list_of_function_arguments = (arithmetic_expression)[BOOST_BIND_ACTION(push_function_arg)] >> *(',' >> (arithmetic_expression)[BOOST_BIND_ACTION(push_function_arg)]);
+            // TODO: should be: "-list_of_function_arguments" but fail to compile
```

Comment:
operator:: - 
is binary ( a - b , match a but not b)
the star(*) operator , is correct , zero or more function arguments

Reply:
i wanted to use the optional operator: https://www.boost.org/doc/libs/1_72_0/libs/spirit/doc/html/spirit/qi/reference/operator/optional.html

Reply:
i see.
the exclamation mark / ! / work fine also. (1 or empty string)

Reply:
will give it a try

Reply:
working